### ISSUE TITLE
Auto-indent makes surprising adjustments when editing lines with temporary syntax errors

### ISSUE THREAD

**Issue by maxbrunsfeld** [zed-industries/feedback#389]
### Check for existing issues

- [X] Completed

### Describe the bug

When there are syntax errors, Zed's auto-indent system sometimes makes annoying indentation adjustments to lines that you are actively editing.

### To reproduce

1. Create a Rust file with the following invalid code:

    ```rs
    fn main() {
        match c.get(3)
    }
    ```
2. Put the cursor on the second line and type backspace
3. :boom: That line is outdented unexpectedly

    ```rs
    fn main() {
    match c.get(3
    }
    ```

### Expected behavior

The line's indentation should not be adjusted.

### Environment

Zed 0.48.1 – /Applications/Zed.app
macOS 12.4
architecture arm64

**Comment by etsea117** [comment]
I am also experiencing this issue

**Comment by chriskrycho** [comment]
Here's a video from a demo I made using Zed earlier this week (which is what inspired #488) which shows the behavior:

https://user-images.githubusercontent.com/2403023/187825421-53dbcee8-082e-4acb-a2a4-c583b3a834ba.mov

**Comment by maxbrunsfeld** [comment]
This fix will go out in Zed 0.71.
